### PR TITLE
translation(JS): Math log, log10, log1p and log2

### PR DESCRIPTION
**Pull request by AdriandeCita**
Original content: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log10, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log1p, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log2

### CODE REVIEW

File: files/uk/web/javascript/reference/global_objects/math/log/index.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Натуральний логарифм (за основою {{jsxref("Math.E", "e")}}) переданого числа. Якщо число від'ємне, буде повернено {{jsxref("NaN")}}.
+
+## Опис
+
+Якщо значення `x` дорівнює 0, повернений результат завжди дорівнює {{jsxref("Number.NEGATIVE_INFINITY", "-Infinity")}}.
+
+Якщо значення `x` — від'ємне, повернений результат завжди дорівнює {{jsxref("NaN")}}.
+
+Оскільки `log()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log()`. Не слід звертатись до нього, як до методу створеного вами об'єкту `Math` (`Math` не є конструктором).
```
Comment (AdriandeCita) [inline files/uk/web/javascript/reference/global_objects/math/log/index.md:65]:
```suggestion
Оскільки `log()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+\= 1.000000000000001, але 1 + 1e-16 = 1.000000000000000, або рівно 1.0 в такій арифметиці, оскільки числа після 15 розряду зникають під час округлення.
+
+Під час обчислення log(1 + x), ми отримуємо дуже близький до `x` результат, за умови, що `x` малий (саме тому його називають 'натуральним' логарифмом). Коли ми викликаємо Math.log(1 + 1.1111111111e-15), ми очікуємо отримати результат, близький до 1.1111111111e-15. Натомість доведеться брати логарифм 1.00000000000000111022 (помилки заокруглення накопичуються у двійковій системі числення, тому інколи вони виглядають незграбно), що дасть нам в результаті 1.11022...e-15, де лише 3 правильні цифри. Якщо натомість обчислити Math.log1p(1.1111111111e-15), отримаємо значно точніший результат 1.1111111110999995e-15, з 15 правильними розрядами точності (в цьому випадку навіть 16).
+
+Якщо значення `x` менше за -1, повернене значення завжди дорівнює {{jsxref("NaN")}}.
+
+Оскільки `log1p()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log1p()`. Не слід звертатись до нього, як до методу створеного вами об'єкту `Math` (`Math` не є конструктором).
```
Comment (AdriandeCita) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:63]:
```suggestion
Оскільки `log1p()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log1p()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

File: files/uk/web/javascript/reference/global_objects/math/log/index.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+### Повернене значення
+
+Натуральний логарифм (за основою {{jsxref("Math.E", "e")}}) переданого числа. Якщо число від'ємне, буде повернено {{jsxref("NaN")}}.
+
+## Опис
+
+Якщо значення `x` дорівнює 0, повернений результат завжди дорівнює {{jsxref("Number.NEGATIVE_INFINITY", "-Infinity")}}.
+
+Якщо значення `x` — від'ємне, повернений результат завжди дорівнює {{jsxref("NaN")}}.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log/index.md:63]:
```suggestion
Якщо значення `x` — від'ємне, повернений результат завжди {{jsxref("NaN")}}.
```
Тут із "дорівнює" треба обережно

File: files/uk/web/javascript/reference/global_objects/math/log/index.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Оскільки `log()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log/index.md:65]:
```suggestion
Оскільки `log()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log()`. Не слід звертатись до нього як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

File: files/uk/web/javascript/reference/global_objects/math/log/index.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+Якщо потрібно отримати натуральний логарифм 2 чи 10, використовуйте сталі {{jsxref("Math.LN2")}} чи {{jsxref("Math.LN10")}}. Якщо ж потрібен логарифм за основою 2 чи 10, для цього існують функції {{jsxref("Math.log2()")}} та {{jsxref("Math.log10()")}}. Для отримання логарифму за якоюсь іншою основою, використайте `Math.log(x) / Math.log(іншаОснова)`, як це показано в прикладі нижче; можливо ви також захочете заздалегідь обчислити значення `1 / Math.log(іншаОснова)`.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log/index.md:67]:
```suggestion
Якщо потрібно отримати натуральний логарифм 2 чи 10 – використовуйте сталі {{jsxref("Math.LN2")}} чи {{jsxref("Math.LN10")}}. Якщо ж потрібен логарифм за основою 2 чи 10 – для цього існують функції {{jsxref("Math.log2()")}} та {{jsxref("Math.log10()")}}. Для отримання логарифму за якоюсь іншою основою – використайте `Math.log(x) / Math.log(іншаОснова)`, як це показано в прикладі нижче; можливо, ви також захочете заздалегідь обчислити значення `1 / Math.log(іншаОснова)`.
```

File: files/uk/web/javascript/reference/global_objects/math/log/index.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+## Приклади
+
+### Застосування Math.log()
+
+```js
+Math.log(-1); // NaN, out of range
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log/index.md:74]:
```suggestion
Math.log(-1); // NaN, вихід за межі області визначення
```

File: files/uk/web/javascript/reference/global_objects/math/log10/index.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+- `x`
+  - : Число.
+
+### Повернене значення
+
+Логарифм переданого числа за основою 10. Якщо число від'ємне, буде повернено {{jsxref("NaN")}}.
+
+## Опис
+
+Якщо значення `x` менше за 0, повернений результат завжди дорівнює {{jsxref("NaN")}}.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log10/index.md:62]:
```suggestion
Якщо значення `x` менше за 0, то повернений результат завжди {{jsxref("NaN")}}.
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+- `x`
+  - : Число.
+
+### Повернене значення
+
+Натуральний логарифм (за основою {{jsxref("Math.E", "e")}}) від **1**, доданої до переданого числа. Якщо число менше за **-1** — буде повернено {{jsxref("NaN")}}.
+
+## Опис
+
+Для дуже малих значень _x_, додавання 1 може зменшити точність. Числа з рухомою комою подвійної точності, які використовуються в JS дають близько 15 розрядів точності. 1 + 1e-15
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:56]:
```suggestion
Для дуже малих значень _x_ додавання 1 може зменшити точність. Числа з рухомою комою подвійної точності, які використовуються в JS, дають близько 15 розрядів точності. 1 + 1e-15
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+\= 1.000000000000001, але 1 + 1e-16 = 1.000000000000000, або рівно 1.0 в такій арифметиці, оскільки числа після 15 розряду зникають під час округлення.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:57]:
```suggestion
\= 1.000000000000001, але 1 + 1e-16 = 1.000000000000000, або рівно 1.0 в такій арифметиці, оскільки цифри після 15 розряду зникають під час округлення.
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+Під час обчислення log(1 + x), ми отримуємо дуже близький до `x` результат, за умови, що `x` малий (саме тому його називають 'натуральним' логарифмом). Коли ми викликаємо Math.log(1 + 1.1111111111e-15), ми очікуємо отримати результат, близький до 1.1111111111e-15. Натомість доведеться брати логарифм 1.00000000000000111022 (помилки заокруглення накопичуються у двійковій системі числення, тому інколи вони виглядають незграбно), що дасть нам в результаті 1.11022...e-15, де лише 3 правильні цифри. Якщо натомість обчислити Math.log1p(1.1111111111e-15), отримаємо значно точніший результат 1.1111111110999995e-15, з 15 правильними розрядами точності (в цьому випадку навіть 16).
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:59]:
```suggestion
Під час обчислення log(1 + x) отримуємо дуже близький до `x` результат, за умови, що `x` малий (саме тому його називають 'натуральним' логарифмом). Коли ми викликаємо Math.log(1 + 1.1111111111e-15), ми очікуємо отримати результат, близький до 1.1111111111e-15. Натомість доведеться брати логарифм 1.00000000000000111022 (помилки заокруглення накопичуються у двійковій системі числення, тому інколи вони виглядають незграбно), що дасть нам в результаті 1.11022...e-15, де лише 3 правильні цифри. Якщо натомість обчислити Math.log1p(1.1111111111e-15), отримаємо значно точніший результат 1.1111111110999995e-15, з 15 правильними розрядами точності (в цьому випадку навіть 16).
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+Якщо значення `x` менше за -1, повернене значення завжди дорівнює {{jsxref("NaN")}}.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:61]:
```suggestion
Якщо значення `x` менше за -1, то повернене значення завжди {{jsxref("NaN")}}.
```

File: files/uk/web/javascript/reference/global_objects/math/log1p/index.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+Оскільки `log1p()` — це статичний метод об'єкта `Math`, його потрібно завжди використовувати через `Math.log1p()`. Не слід звертатись до нього, як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log1p/index.md:63]:
```suggestion
Оскільки `log1p()` — це статичний метод об'єкта `Math` – його потрібно завжди використовувати через `Math.log1p()`. Не слід звертатись до нього як до методу власноруч створеного екземпляра `Math` (`Math` не є конструктором).
```

File: files/uk/web/javascript/reference/global_objects/math/log2/index.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+- `x`
+  - : Число.
+
+### Повернене значення
+
+Логарифм переданого числа за основою 2. Якщо передано від'ємне число, буде повернено {{jsxref("NaN")}}.
+
+## Опис
+
+Якщо значення `x` менше за 0, повернене значення завжди дорівнює {{jsxref("NaN")}}.
```
Comment (undead404) [inline files/uk/web/javascript/reference/global_objects/math/log2/index.md:62]:
```suggestion
Якщо значення `x` менше за 0, то повернене значення завжди {{jsxref("NaN")}}.
```

**Review by undead404 — CHANGES_REQUESTED** [review]

**Review by undead404 — APPROVED** [review]

**Pull request merged**